Homework_RalucaHotea/CodeChallenge3/app/src/main/java/com/example/hommework3/SecondActivity.java

```diff
@@ -0,0 +1,35 @@
+package com.example.hommework3;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.CompoundButton;
+import android.widget.Spinner;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+public class SecondActivity extends AppCompatActivity implements AdapterView.OnItemSelectedListener{
+    String versions[] = {"Cupcake", "Donut", "Eclair", "KitKat"};
+    Spinner spinner = (Spinner) findViewById(R.id.android_spinner);
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.my_spinner_view);
+
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this,
+                R.array.planets_array, android.R.layout.simple_spinner_item);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(adapter);
+    }
+    @Override
+    public void onItemSelected(AdapterView<?> arg0, View arg1, int position, long id) {
+        Toast.makeText(getApplicationContext(),versions[position] , Toast.LENGTH_LONG).show();
```

Comment:
Please use the activity context, not the application one. They are both instances of Context, but the application instance is tied to the lifecycle of the application, while the Activity instance is tied to the lifecycle of an Activity. Thus, they have access to different information about the application environment. In general, use the activity context unless you have a good reason not to.